features/FEATURE_PSA/supporting_psa_in_mbed-os.md

```diff
@@ -0,0 +1,202 @@
+# Supporting PSA in Mbed OS
+
+This document is still work in progress. It describes the process of adding new PSA targets to Mbed OS. It also outlines the new directory structure created to support PSA on single v7-M, dual v7-M and single v8-M.
+
+## TODO
+* Adding dual v7-M targets
+* Adding v8-M targets
+* Building TF-M
+* Running PSA compliance tests
+
+## Adding new PSA targets
+To help with the creation of new PSA targets, couple of generic PSA targets have been added to `targets/targets.json`.
+* `PSA_Target` (Root level PSA target)
+* `PSA_V7_M_NSPE` (Single v7-M NSPE generic target)
+* `PSA_V7_M_SPE` (Single v7-M SPE generic target)
+* `PSA_DUAL_V7_M_NSPE` (Dual v7-M NSPE generic target)
+* `PSA_DUAL_V7_M_SPE` (Dual v7-M SPE generic target)
+* `PSA_V8_M_NSPE` (v8-M NSPE generic target)
+* `PSA_V8_M_SPE` (v8-M SPE generic target)
+
+The new PSA targets **MUST** `inherit` from one of these targets depending on the architecture:
+
+Example: 
+```json
+    "K64F": {
+        "inherits": ["PSA_V7_M_NSPE"],
+    }
+```
+
+The generic PSA target `PSA_Target` inherits from `Target` and other generic PSA targets inherits from `PSA_Target`, therefore it is **NOT** needed to add 
+```json
+"inherits": ["Target"],
+```
+to PSA target definition.
+
+### Adding single v7-M targets
```

Comment:
@Patater Good catch. The method described in the document should be used while creating new PSA targets. Because, just adding `"features_add": ["PSA"]` won't bring in PSA services for v7-M (`"extra_labels": ["MBED_PSA_SRV"]`). I've updated targets.json.